plugins/completion/nvim-cmp/default.nix

```diff
@@ -115,103 +79,157 @@ in {
       '';
     };
 
-    completion = mkOption {
-      default = null;
-      type = types.nullOr (types.submodule ({...}: {
-        options = {
-          keyword_length = mkOption {
-            default = null;
-            type = types.nullOr types.int;
-          };
+    mappingPresets = mkOption {
+      default = [];
+      type = types.listOf (types.enum [
+        "insert"
+        "cmdline"
+      ]);
+      description = ''
+        Mapping presets to use; cmp.mapping.preset.
+        \$\{mappingPreset} will be called with the configured mappings.
+      '';
+      example = ''[ "insert" "cmdline" ]'';
+    };
 
-          keyword_pattern = mkOption {
-            default = null;
-            type = types.nullOr types.str;
-          };
+    snippet = helpers.mkCompositeOption "Snippet options" {
+      expand =
+        helpers.defaultNullOpts.mkStr
+        ''
+          function(_)
+            error('snippet engine is not configured.')
+          end
+        ''
+        ''
+          The snippet expansion function. That's how nvim-cmp interacts with a
+          particular snippet engine.
+
+          Example:
+          ```
+          function(args)
+            vim.fn["vsnip#anonymous"](args.body) -- For `vsnip` users.
+            -- require('luasnip').lsp_expand(args.body) -- For `luasnip` users.
+            -- require('snippy').expand_snippet(args.body) -- For `snippy` users.
+            -- vim.fn["UltiSnips#Anon"](args.body) -- For `ultisnips` users.
+          end
+          ```
+        '';
+    };
 
-          autocomplete = mkOption {
-            default = null;
-            type = types.nullOr types.str;
-            description = "Lua code for the event.";
-            example = ''"false"'';
-          };
+    completion = helpers.mkCompositeOption "Completion options" {
+      keywordLength = helpers.defaultNullOpts.mkInt 1 ''
+        The number of characters needed to trigger auto-completion.
+      '';
 
-          completeopt = mkOption {
-            default = null;
-            type = types.nullOr types.str;
-          };
-        };
-      }));
+      keywordPattern =
+        helpers.defaultNullOpts.mkStr
+        "[[\%(-\?\d\+\%(\.\d\+\)\?\|\h\w*\%(-\w*\)*\)]]"
+        "The default keyword pattern.";
```

Comment:
Same tretament as todo-comments, mkRaw with [[ ]] appendended, and  a note in the documentation

Reply:
Done :)